vue/sbc-common-components/src/components/SbcSignin.vue

```diff
@@ -36,34 +38,50 @@ import { KCUserProfile } from '../models/KCUserProfile'
     }
   }
 })
-export default class SbcSignin extends Vue {
+export default class SbcSignin extends NavigationMixin {
   private isLoading = true
   @Prop({ default: 'bcsc' }) idpHint!: string
   @Prop({ default: '' }) redirectUrlLoginFail!: string
+  @Prop({ default: false }) inAuth!: boolean;
   private readonly loadUserInfo!: () => KCUserProfile
   private readonly syncAccount!: () => Promise<void>
 
   private async mounted () {
     getModule(AccountModule, this.$store)
     // Initialize keycloak session
-    const kcInit = await this.initKeycloak(this.idpHint)
+    const kcInit = await KeyCloakService.initializeKeyCloak(this.idpHint, this.$store)
     await new Promise((resolve, reject) => {
       kcInit.success(async (authenticated: boolean) => {
         if (authenticated) {
           // Set values to session storage
-          KeyCloakService.initSession()
-          // emitting event for the header to get updated with :key increment from the parent component
-          this.$emit('keycloak-session-ready')
+          await KeyCloakService.initSession()
           // tell KeycloakServices to load the user info
-          this.loadUserInfo()
+          const userInfo = await this.loadUserInfo()
           // sync the account if there is one
           await this.syncAccount()
-          this.$emit('sync-user-profile-ready')
           // eslint-disable-next-line no-console
           console.info('[SignIn.vue]Logged in User.Starting refreshTimer')
-          let tokenService = new TokenService()
-          await tokenService.init()
-          tokenService.scheduleRefreshTimer()
+          await KeyCloakService.initializeToken(this.$store)
```

Comment:
It looks like some unwanted calls here. There is initializeKeyCloak and then initializeToken. And initializeToken again does a keycloak init. Can we simplify this for better maintenance ?